connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -430,7 +432,9 @@ public Void call() throws Exception {
                         if (!configState.contains(connName)) {
                             callback.onCompletion(new NotFoundException("Connector " + connName + " not found"), null);
                         } else {
-                            callback.onCompletion(null, new ConnectorInfo(connName, configState.connectorConfig(connName), configState.tasks(connName)));
+                            callback.onCompletion(null, new ConnectorInfo(connName, configState.connectorConfig(connName),
+                                configState.tasks(connName),
+                                WorkerConnector.getConnectorType(configState.connectorConfig(connName).get(ConnectorConfig.CONNECTOR_CLASS_CONFIG))));
```

Comment:
I'm afraid I might have led you down a rabbit hole. The `Worker.isSinkConnector(String name)` uses the Plugin mechanism to properly load the class with the correct classloader. My recent request to move that logic into the WorkerConnector would simplify the logic, exception in a case like this where we don't have a WorkerConnector instance. So perhaps the best thing is to use the new `worker.getConnectorType(className)` that basically just uses the `worker.isSinkConnector(className)`.

Apologies for causing you extra work. It would have been nice for my suggestion to work out, but its the case where that isn't known that is causing the difficulties. Using the worker should always work (assuming the connector is installed to begin with).

Reply:
This code is also getting the connector class name twice. Probably be better to do that once before calling the `onCompletion(...)` method.

Reply:
Without Connector instance, the following cannot be applied:
```
            savedLoader = plugins.compareAndSwapLoaders(workerConnector.connector());
```
...